imap_processing/idex/idex_l1a.py

```diff
@@ -355,11 +307,11 @@ def _set_sample_trigger_times(
         rather than the number of samples before triggering.
         """
         # Retrieve the number of samples of high gain delay
-        high_gain_delay = packet["IDX__TXHDRADC0IDELAY"]
-
+        high_gain_delay = (packet["IDX__TXHDRSAMPDELAY"] >> 20) & 0b1111111111
+        n_blocks = packet["IDX__TXHDRBLOCKS"]
         # Retrieve number of low/high sample pre-trigger blocks
-        num_low_sample_pretrigger_blocks = packet["IDX__TXHDRLSPREBLOCKS"]
-        num_high_sample_pretrigger_blocks = packet["IDX__TXHDRHSPREBLOCKS"]
+        num_low_sample_pretrigger_blocks = (n_blocks >> 6) & 0b111111
+        num_high_sample_pretrigger_blocks = (n_blocks >> 16) & 0b1111
```

Comment:
Could you explain in a comment what is going on here? It is a bit confusing as a reader.